data_diff/database.py

```diff
@@ -66,12 +70,77 @@ def _query_conn(conn, sql_code: str) -> list:
     return c.fetchall()
 
 
-class Database(ABC):
+class ColType:
+    pass
+
+
+@dataclass
+class PrecisionType(ColType):
+    precision: Optional[int]
+
+
+class TemporalType(PrecisionType):
```

Comment:
I think `TemporalType` also would then include `Time`, `Date`, and `Interval` eventually, which are a bit different. We'll proper need another layer of class between this over-arching time type and `Timestamp`, `TimestampTZ`, and `Datetime`

Reply:
Yeah, maybe. Not sure how to call it though.